scanpipe/pipes/resolve.py

```diff
@@ -125,6 +126,76 @@ def resolve_spdx_packages(input_location):
     ]
 
 
+def cyclonedx_component_to_discovered_package_data(component_data):
+    extra_data = {}
+    component = component_data["cdx_package"]
+
+    package_url_dict = (
+        PackageURL.from_string(component.purl).to_dict(encode=True)
+        if component.purl
+        else {}
+    )
+
+    checksum_data = cyclonedx.get_checksums(component)
+    external_references = cyclonedx.get_external_refrences(component.externalReferences)
+    homepage_url = (
+        external_references["website"][0] if "website" in external_references else ""
+    )
+    bug_tracking_url = (
+        external_references["issue-tracker"][0]
+        if "issue-tracker" in external_references
+        else ""
+    )
+    vcs_url = external_references["vcs"][0] if "vcs" in external_references else ""
+
+    if external_references:
+        extra_data["externalReferences"] = external_references
+    if component_data["nested_components"]:
+        extra_data["nestedComponents"] = component_data["nested_components"]
+
+    package_data = {
+        "name": component.name,
+        "declared_license": cyclonedx.get_declared_licenses(
+            list_of_license_obj=component.licenses
+        ),
+        "copyright": component.copyright,
+        "version": component.version,
+        "description": component.description,
+        "homepage_url": homepage_url,
+        "bug_tracking_url": bug_tracking_url,
+        "vcs_url": vcs_url,
+        "extra_data": extra_data,
+        **package_url_dict,
+        **checksum_data,
+    }
+
+    return {
+        key: value for key, value in package_data.items() if value not in [None, ""]
```

Comment:
```suggestion
        key: value for key, value in package_data.items() if not value
```
Probably this is enough?

Reply:
> Probably this is enough?

Yes